kong/plugins/http-log/schema.lua

```diff
@@ -8,15 +10,14 @@ return {
     { config = {
         type = "record",
         fields = {
-          -- NOTE: any field added here must be also included in the handler's get_queue_id method
           { http_endpoint = typedefs.url({ required = true, encrypted = true, referenceable = true }) }, -- encrypted = true is a Kong-Enterprise exclusive feature, does nothing in Kong CE
           { method = { type = "string", default = "POST", one_of = { "POST", "PUT", "PATCH" }, }, },
           { content_type = { type = "string", default = "application/json", one_of = { "application/json", "application/json; charset=utf-8" }, }, },
           { timeout = { type = "number", default = 10000 }, },
           { keepalive = { type = "number", default = 60000 }, },
-          { retry_count = { type = "integer", default = 10 }, },
-          { queue_size = { type = "integer", default = 1 }, },
-          { flush_timeout = { type = "number", default = 2 }, },
+          { retry_count = { type = "integer" }, },
+          { queue_size = { type = "integer" }, },
+          { flush_timeout = { type = "number" }, },
```

Comment:
@hbagdi, this is changing defaults. Recently I remembered that you were against this, was it in session PR: https://github.com/Kong/kong/pull/10333#issuecomment-1442261133

This seems to be trying to do the same as I initially tried in session PR.

Reply:
I'm not sure if this is the same issue as the quoted one.  We're removing the defaults for these deprecated parameters and we have heuristics to detect whether the old parameters are used (i.e. if they're set to a value that is different from the old default, we generate deprecation warning).  Our expectation is that these parameters will eventually not be used anymore.

Reply:
@hbagdi Can you please comment?  Is it OK to _remove_ defaults in preparation of removing them?

Reply:
Thanks for bringing this to my attention.
It is unfortunate that all of these changes are very nuanced and a little bit different.

1. From what I gather, these parameters are now useless. They are not deprecated, they are just useless. And we need to be accurate in our usage of those words. Deprecation means something still works and will be removed in the future. Useless means these parameters won't work. Am I missing something?
2. Relying on the heuristic of non-default value is okay and is the only way out as I see it. 
3. Removing defaults or changing defaults must not be done generally since it causes serious cross-version compatibility issues. In this specific case, I recommend not removing defaults to adhere to our principled approach, even though I understand that having this defaults or not makes little to no difference since the parameters are useless.

Reply:
@hbagdi if they are useless, they should be just `shorthands` that log the warning about them being useless. But it seems they are still used.

Reply:
Let me also add that even if we just added the new parameters, kept the old ones and would not provide an automatric migration path, the chances that something actually breaks in customer installations is very low.  Let's assume that some customer tuned these underdocumented queue parameters in response to some operational issue that they had at some point.  How likely is it that something would break when these parameter changes are not retained accross a version upgrade?  We have little to no systematic information about the behavior of the queues in production other than knowing that if things go wrong, we run out of memory.

It is not my decision to make, but I think we're taking "no breaking changes" pretty far, and too far in this case.

Reply:
To be clear, we cannot remove these parameters. That would be too big of a breaking change.
Only keeping or removing the default or removing them is in discussion here.

I would like to change my stance and I think we should remove the defaults in this case.
The code must expect that the values are set (they will be for all users who upgrade) and those settings may be the same defaults or new defaults.

I don't think we should do shortahands because the semantics of the configuration options have changed. Doing anything clever can result in disasters that are hard to explain.